src/labelle/lib/constants.py

```diff
@@ -52,36 +53,110 @@ class SUPPORTED_DEVICE_ID(Enum):
     MOBILE_LABELER = 0x1009
 
 
-# fmt: off
-# Very bad I know, but it keeps a loop of removing the line breaks and errroring
+# Create supported products list
+SUPPORTED_PRODUCTS = []
 
-SUPPORTED_PRODUCTS = {
-    SUPPORTED_DEVICE_ID.LABELMANAGER_PC:
+# ---- Supported USB Devices configuration ----
+SUPPORTED_PRODUCTS.append(
```

Comment:
I think that eventually we should move these out of the code, into a JSON file. If possible, please try to do it. In not, we can try refactor it later.

Reply:
I think we need a proper data model(1) for device models(2). (The first "model" in that sentence means "implementation of a schema" while the second "model" refers to a particular product that is sold by a company.)

Reply:
Sorry, I was a bit confused You do already define `DeviceConfig`.

I'm wondering how to achieve this elegantly. I agree that it's in principle better to define these things out-of-code like in JSON. However, that unfortunately would require deserialization. Pydantic is a popular library for this, but then we're adding more project dependencies and a lot of complexity, also in the code.

The way I see it, we have two main options:
1. Separate the device definition from code, perform deserialization, probably add a dependency on something like Pydantic.
2. Keep the device definitions as code, but switch to `NamedTuple` or `dataclass`.

Another disadvantage to 1. is that editing the config would involve editing serialized data which can be a real pain. An advantage to 2. is that code tools like mypy will automatically perform validation for the data-as-code.

I'm leaning more towards 2.

Reply:
What I was trying to achieve here is a what in C would be "array of structs" for the "const" configuration.
If you have any suggestions how to do this elegantly in python, let me know...

My idea was that the (existing) labeler class took this as input to know the device's exact properties.

I agree that having this out of code would be nice. But can we please keep this contained?
I started out with "just adding my printer", and it is getting a bit out of hand already imho.

Reply:
Converted DeviceConfig to a dataclass